microsoft/terminal
# Added CopyOnSelect as a Global Setting

## Summary of the Pull Request
Introduces the "copyOnSelect" setting to "globals" as a boolean type. When set to `true`, a selection is immediately copied to your clipboard upon creation. When set to `false`, the selection persists and awaits further action.



## PR Checklist
* [x] Closes #524 
* [x] Tests ~added/passed~ not necessary
* [x] I am a core contributor


## Detailed Description of the Pull Request / Additional comments
Standard "added a new global setting" changes
`TermControl`'s `PointerReleased` event is updated to support a mouse button release event. Just hooked up the copy function right there.


## Validation Steps Performed
- `copyOnSelect` = true
  - make selection
  - right click to paste (selection persists)
  - Works! ✔
- `copyOnSelect` = true
  - make selection
  - right click to paste
  - left click on a cell
  - right click to paste
  - same data should be pasted twice
  - Works! ✔
- `copyOnSelect` = true
  - click on a cell (no selection)
  - drag to another cell (selection visible)
  - drag back to first cell (selection should persist, but should only cover one cell)
  - Works! ✔
- `copyOnSelect` = false
  - make selection
  - selection persists
  - right click to copy (selection disappears)
  - right click to paste
  - Works! ✔
